File: C++/Algorithms/Recursion/greatestCommonDivisor.cpp
/**
 * Calculates the greatest common divisor of two numbers using recursion
 *
 * @param int firstNumber
 * @param int secondNumber
 * @return int
 */
 int greatestCommonDivisor(int firstNumber, int secondNumber) {
    if (firstNumber == 0) {
        return secondNumber;
    }

    return greatestCommonDivisor(secondNumber % firstNumber, firstNumber);
}

File: PHP/Algorithms/Recursion/greatestCommonDivisor.php
<?php

/**
 * Calculates the greatest common divisor of two numbers using recursion
 *
 * @param int $firstNumber
 * @param int $secondNumber
 * @return int
 */
function greatestCommonDivisor($firstNumber, $secondNumber) {
    if ($firstNumber == 0) {
        return $secondNumber;
    }

    return greatestCommonDivisor($secondNumber % $firstNumber, $firstNumber);
}
